spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/CaffePersisterSpec.scala

```diff
@@ -0,0 +1,117 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils
+
+import java.nio.file.Paths
+
+import com.google.protobuf.GeneratedMessage
+import com.intel.analytics.bigdl.models.resnet.Convolution
+import com.intel.analytics.bigdl.nn.Graph.ModuleNode
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import org.scalatest.{FlatSpec, Matchers}
+import com.intel.analytics.bigdl.numeric.NumericDouble
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.caffe.{CaffeLoader, CaffePersister}
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+import scala.util.Random
+
+
+class CaffePersisterSpec extends FlatSpec with Matchers{
+
+  val resource = getClass().getClassLoader().getResource("caffe")
+  val prototxt = Paths.get(resource.getPath(), "test.prototxt").toString
+  val modelPath = Paths.get(resource.getPath(), "test.caffemodel").toString
+
+  val savedprototxt = Paths.get(resource.getPath(), "test_persist.prototxt").toString
+  val savedmodelPath = Paths.get(resource.getPath(), "test_persist.caffemodel").toString
+
+  private def loadDummy[T : ClassTag](message : GeneratedMessage)(implicit ev: TensorNumeric[T])
+  : Seq[ModuleNode[T]] = {
+    Seq(Identity[T].setName("Dummy").inputs())
+  }
+
+  val convertMap = new mutable.HashMap[String, (GeneratedMessage) => Seq[ModuleNode[Double]]]()
+  convertMap("DUMMY") = loadDummy[Double]
+
+  "Save graph module" should "Works properly" in {
+
+    val convolution1 = new ModuleNode(Convolution(3, 4, 2, 2).
+      setName("conv1").asInstanceOf[AbstractModule[Activity, Tensor[Double], Double]])
+
+    val convolution2 = new ModuleNode(Convolution(4, 3, 2, 2).setName("conv2")
+      .asInstanceOf[AbstractModule[Activity, Tensor[Double], Double]])
+
+    val view = new ModuleNode(View(27).setName("view")
+      .asInstanceOf[AbstractModule[Activity, Tensor[Double], Double]])
+
+    val ip = new ModuleNode(Linear(2, 27, withBias = false).setName("ip")
+      .asInstanceOf[AbstractModule[Activity, Tensor[Double], Double]])
+
+    convolution1 -> convolution2
+
+    convolution2 -> view
+
+    view -> ip
+
+    val module = Graph(convolution1, ip)
+
+    CaffePersister.persist("/tmp/test.prototxt", "/tmp/test.caffemodel", module, overwrite = true)
+
+  }
+
+  "Persist V1 module" should "works properly" in {
+    val module = new ModuleNode(Linear[Double](100, 10).
+      setName("simple linear").asInstanceOf[AbstractModule[Activity, Tensor[Double], Double]])
+    val graph = Graph(module, module)
+    CaffePersister.persist("/tmp/v1.prototxt", "/tmp/v1.caffemodel",
+      graph, useV2 = false, overwrite = true)
+  }
+
```

Comment:
add test for v2 save/load, and you also need hdfs test (like this https://github.com/intel-analytics/BigDL/blob/0c8b8437d8d353dc56e1559afe63de7a09e02a5d/spark/dl/src/test/scala/com/intel/analytics/bigdl/integration/HdfsSpec.scala)